components/common/FormFields.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { ChangeEventHandler } from "react"
+import {
+    Button,
+    Checkbox,
+    Flex,
+    FormControl,
+    FormErrorMessage,
+    FormLabel,
+    Input,
+} from "@chakra-ui/react"
+import styled from "@emotion/styled"
+import ErrorAlert from "./ErrorAlert"
+
+type TaskFieldInputProps = {
+    label: string
+    placeholder: string
+    isRequired: boolean
+    isInvalid?: string
+    formErrorMessage?: string
+    onChange: ChangeEventHandler<HTMLInputElement>
+}
+
+export function FormInputField({
+    label,
+    placeholder,
+    isRequired,
+    isInvalid,
+    formErrorMessage,
+    onChange,
+}: TaskFieldInputProps): JSX.Element {
+    if (isRequired) {
```

Comment:
Is this necessary? I think you should be able to render as follows:

```typescript
<FormControl isInvalid={!isInvalid} isRequired={isRequired}>
```

and if `isRequired` is false or undefined, the component works as intended.

Reply:
![Screenshot from 2022-02-08 09-51-11](https://user-images.githubusercontent.com/17109116/152941986-561e9a38-9704-47df-bc4d-d519a8bbc22f.png)

Reply:
It will work, but the isInvalid will always think the description field is invalid. Because isRequired is false on the desciription field, the form will still go through but I don't think it is intended behavior

Reply:
Yes this is not intended.  implementing my suggestions below should fix this.